src/jsm.c

```diff
@@ -3446,6 +3452,140 @@ js_DeleteConsumer(jsCtx *js, const char *stream, const char *consumer,
     return NATS_UPDATE_ERR_STACK(s);
 }
 
+natsStatus
+js_unmarshalConsumerPauseResp(nats_JSON *json, jsConsumerPauseResponse **new_cpr)
+{
+    natsStatus s = NATS_OK;
+    jsConsumerPauseResponse *cpr = NULL;
+
+    cpr = (jsConsumerPauseResponse *)NATS_CALLOC(1, sizeof(jsConsumerPauseResponse));
+    if (cpr == NULL)
+        return nats_setDefaultError(NATS_NO_MEMORY);
+
+    s = nats_JSONGetBool(json, "paused", &(cpr->Paused));
+    IFOK(s, nats_JSONGetTime(json, "pause_until", &(cpr->PauseUntil)));
+    IFOK(s, nats_JSONGetLong(json, "pause_remaining", &(cpr->PauseRemaining)));
+
+    if (s == NATS_OK)
+        *new_cpr = cpr;
+    else
+        NATS_FREE(cpr);
+
+    return NATS_UPDATE_ERR_STACK(s);
+}
+
+static natsStatus
+_unmarshalConsumerPauseResp(jsConsumerPauseResponse **new_cpr, natsMsg *resp, jsErrCode *errCode)
+{
+    nats_JSON *json = NULL;
+    jsApiResponse ar;
+    natsStatus s;
+
+    s = js_unmarshalResponse(&ar, &json, resp);
+    if (s != NATS_OK)
+        return NATS_UPDATE_ERR_STACK(s);
+
+    if (js_apiResponseIsErr(&ar))
+    {
+        if (errCode != NULL)
+            *errCode = (int)ar.Error.ErrCode;
+
+        if (ar.Error.ErrCode == JSConsumerNotFoundErr)
+            s = NATS_NOT_FOUND;
+        else
+            s = nats_setError(NATS_ERR, "%s", ar.Error.Description);
+    }
+    else if (new_cpr != NULL)
+    {
+        // At this point we need to unmarshal the consumer info itself.
+        s = js_unmarshalConsumerPauseResp(json, new_cpr);
+    }
+
+    js_freeApiRespContent(&ar);
+    nats_JSONDestroy(json);
+
+    return NATS_UPDATE_ERR_STACK(s);
+}
+
+static natsStatus
+_marshalConsumerPauseReq(natsBuffer **new_buf, uint64_t pauseUntil)
+{
+    natsStatus s = NATS_OK;
+    natsBuffer *buf = NULL;
+
+
+    s = natsBuf_Create(&buf, 256);
+    IFOK(s, natsBuf_AppendByte(buf, '{'));
+    if ((s == NATS_OK) && (pauseUntil > 0)) {
+        s = _marshalTimeUTC(buf, false, "pause_until", pauseUntil);
+    }
+    IFOK(s, natsBuf_AppendByte(buf, '}'));
+
+    if (s == NATS_OK)
+        *new_buf = buf;
+    else
+        natsBuf_Destroy(buf);
+
+    return NATS_UPDATE_ERR_STACK(s);
+}
+
+natsStatus
+js_PauseConsumer(jsConsumerPauseResponse **new_cpr, jsCtx *js,
```

Comment:
I would add a jsConsumerPauseResponse_Destroy(), which currently would be a simple `NATS_FREE` of the pointer, but will be more portable.

Reply:
done.